### ISSUE TITLE
eslint-disable-next-line is ignored

### ISSUE THREAD

**Issue by Telokis** [eslint/eslint#8618]
**Tell us about your environment**

* **ESLint Version:** "eslint": "^3.19.0"
* **Node Version:** v5.7.0
* **npm Version:** 3.5.3

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**
```
{
    "extends": "airbnb-base",
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module"
    },
    "rules": {
        // Customized
        "handle-callback-err": ["error", "^(e$|(e|(.*(_e|E)))rr)"],
        "comma-dangle": ["error", "never"],
        "no-empty": ["error", { "allowEmptyCatch": true }],
        "no-underscore-dangle": "off",
        "newline-per-chained-call": "warn",
        "no-console": "warn",
        "no-mixed-operators": ["error", { "allowSamePrecedence": true }],
        "strict": ["error", "global"],
        "consistent-return": "off",
        "func-names": "off",
        "no-tabs": "error",
        "indent": ["error", 4],
        "no-eq-null": "warn",
        "camelcase": "warn",
        "no-new": "warn",
        "no-shadow": "off",
        "no-use-before-define": ["error", "nofunc"],
        "no-prototype-builtins": "warn",
        "new-cap": "warn",
        "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }],
        "import/no-unresolved": "error",

        // ES6
        "prefer-rest-params": "warn",
        "prefer-spread": "warn",
        "prefer-arrow-callback": "warn",
        "prefer-template": "warn",
        "no-var": "error",
        "object-shorthand": "warn",
        "vars-on-top": "warn",

        // TODO
        "import/no-extraneous-dependencies": "off",
        "import/no-dynamic-require": "warn",
        "import/newline-after-import": "warn",
        "no-bitwise": "warn",
        "global-require": "off",
        "max-len": "warn",
        "no-param-reassign": "off",
        "no-restricted-syntax": "warn",
        "no-script-url": "warn",
        "default-case": "warn",

        "no-multi-assign": "warn",
        "linebreak-style": ["error", "unix"],
        "one-var": "warn",
        "no-undef": "warn",
        "max-nested-callbacks": "warn",
        "no-mixed-requires": "warn",
        "brace-style": "warn",
        "max-statements-per-line": "warn",
        "no-unused-vars": "warn",
        "no-mixed-spaces-and-tabs": "warn",
        "no-useless-concat": "warn",
        "require-jsdoc": ["error", {
            "require": {
                "FunctionDeclaration": true,
                "MethodDefinition": true,
                "ClassDeclaration": true,
                "ArrowFunctionExpression": true
            }
        }],
        "eqeqeq": "warn",
        "no-negated-condition": "off",
        "one-var-declaration-per-line": "warn",
        "class-methods-use-this": "off",
        "no-lonely-if": "warn",
        "radix": "warn",
        "no-else-return": "warn",
        "no-useless-escape": "warn",
        "block-scoped-var": "warn",
        "operator-assignment": "warn",
        "yoda": "warn",
        "no-loop-func": "warn",
        "no-void": "warn",
        "valid-jsdoc": ["error", {
            "requireReturn": false,
            "requireParamDescription": false
        }],
        "no-cond-assign": "warn",
        "no-redeclare": "warn",
        "no-unreachable": "warn",
        "no-nested-ternary": "warn",
        "operator-linebreak": "warn",
        "guard-for-in": "warn",
        "no-unneeded-ternary": "warn",
        "no-sequences": "warn",
        "no-extend-native": "warn",
        "no-shadow-restricted-names": "warn",
        "no-extra-boolean-cast": "warn",
        "no-path-concat": "warn",
        "no-unused-expressions": "warn",
        "no-return-assign": "warn",
        "no-restricted-modules": "warn",
        "object-curly-spacing": "warn",
        "padded-blocks": "warn",
        "eol-last": "warn",
        "lines-around-directive": "warn",
        "no-multi-spaces": "warn",
        "quotes": "warn",
        "keyword-spacing": "warn",
        "comma-spacing": "warn",
        "no-trailing-spaces": "error",
        "key-spacing": [
            "warn",
            {
                "align": {
                    "beforeColon": true,
                    "afterColon": true,
                    "on": "colon"
                }
            }
        ],
        "no-multiple-empty-lines": "warn",
        "spaced-comment": "warn",
        "space-in-parens": "warn",
        "block-spacing": "warn",
        "quote-props": "warn",
        "space-unary-ops": "warn",
        "dot-notation": "warn",
        "func-call-spacing": "warn",
        "array-bracket-spacing": "warn",
        "object-property-newline": "warn",
        "no-continue": "warn",
        "no-extra-semi": "warn",
        "no-spaced-func": "warn",
        "no-useless-return": "warn"
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
import express from 'express';
import path from 'path';
import compression from 'compression';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { match, RouterContext } from 'react-router';
import routes from '../client/Routes';
import * as nconf from 'nconf';

const app = express();

app.use(compression());
app.use(express.static(path.join(__dirname, "/src/public")));

app.get('*', (req, res) => {
    match({ routes, location: req.url }, (err, redirect, props) => {
        if (err) {
            res.status(500).send(err.message);
        } else if (redirect) {
            res.redirect(redirect.pathname + redirect.search);
        } else if (props) {
            //eslint-disable-next-line
            const htmlData = renderToString(<RouterContext {...props } />);
            res.send(htmlData);
        }
        else {
            res.status(404).send('Not Found');
        }
    });
});

const PORT = nconf.get('port') || 8080;
app.listen(PORT, () => {
    console.log('Express server running on port ' + PORT);
});
```

**What did you expect to happen?**
The line `const htmlData = renderToString(<RouterContext {...props } />);` should be ignored

**What actually happened? Please include the actual, raw output from ESLint.**
`23:45  error  Parsing error: Unexpected token <`

Maybe I should use the react parser but this is the only file where I'd like to apply it. Is there a way to do so?

**Comment by platinumazure** [comment]
You need to set your parserOptions as follows:

```json
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    }
}
```

As for eslint-disable-next-line not working, unfortunately we do have to parse the whole file before we can read any comments for inline configuration directives. I believe this is covered in our Configuration documentation but would have to double check.

**Comment by Telokis** [comment]
Is there a way to specify those parser options inside the file so that it is the only one affected?

**Comment by not-an-aardvark** [comment]
No, because ESLint would have to start parsing the file in order to find any directives in the file.
